src/main/java/seedu/address/model/transaction/Quantity.java

```diff
@@ -3,33 +3,152 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import seedu.address.logic.parser.exceptions.ParseException;
+
 /**
  * Class to store the quantity of goods transacted.
  */
 public class Quantity {
-    public static final String MESSAGE_CONSTRAINTS =
-            "Quantity should only contain numbers, and it should be at least 1 digits long";
+
     public static final String VALIDATION_REGEX = "\\d{1,}";
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Quantity should only contain a positive integer.";
+
+    public static final String MESSAGE_CONSTRAINTS_REGEX =
+            "Quantity should not contain decimal point.";
+
+    public static final String MESSAGE_CONSTRAINTS_ZERO =
+            "Quantity should not be 0.";
+    public static final String MESSAGE_CONSTRAINTS_POSITIVE =
+            "Quantity should not be be negative.";
+
+    public static final String MESSAGE_CONSTRAINTS_LARGE =
+            "Quantity should be less than 1 million.";
+
+    public static final String MESSAGE_CONSTRAINTS_EMPTY = "Quantity should not be left empty.";
+
```

Comment:
Perhaps you could clean up this portion of the code if you're not using it anymore?